f5_openstack_agent/lbaasv2/drivers/bigip/lbaas_builder.py

```diff
@@ -134,186 +158,120 @@ def _assure_listeners_created(self, service):
-                        self.pool_builder.create_pool(svc, bigips)
-                    else:
-                        self.pool_builder.update_pool(svc, bigips)
-                except HTTPError as err:
-                    if err.response.status_code != 409:
-                        pool['provisioning_status'] = plugin_const.ERROR
-                        loadbalancer['provisioning_status'] = \
-                            plugin_const.ERROR
-                        raise f5_ex.PoolCreationException(err.message)
-                except Exception as err:
-                    pool['provisioning_status'] = plugin_const.ERROR
-                    loadbalancer['provisioning_status'] = \
-                        plugin_const.ERROR
-                    raise f5_ex.PoolCreationException(str(err))
-
-    def _assure_pools_configured(self, service):
-        if "pools" not in service:
-            return
-
-        pools = service["pools"]
-        loadbalancer = service["loadbalancer"]
-
-        bigips = self.driver.get_config_bigips()
-
-        for pool in pools:
-            if pool['provisioning_status'] != plugin_const.PENDING_DELETE and \
-                    self._is_not_error(pool):
-                svc = {"loadbalancer": loadbalancer,
-                       "pool": pool}
-                svc['members'] = self._get_pool_members(service, pool['id'])
-                try:
-                    # assign pool name to virtual
-                    pool_name = self.service_adapter.init_pool_name(
-                        loadbalancer, pool)
-
-                    # get associated listeners for pool
-                    listeners = self._get_pool_listeners(service, pool['id'])
-                    for listener in listeners:
-                        svc['listener'] = listener
-                        self.listener_builder.update_listener_pool(
-                            svc, pool_name["name"], bigips)
-
-                        # update virtual sever pool name, session persistence
-                        self.listener_builder.update_session_persistence(
-                            svc, bigips)
-                    self._set_status_as_active(pool)
-                except Exception as err:
+                error = self.pool_builder.create_pool(svc, bigips)
+                if error:
                     pool['provisioning_status'] = plugin_const.ERROR
                     loadbalancer['provisioning_status'] = plugin_const.ERROR
-                    raise f5_ex.PoolCreationException(err.message)
-
-    def _get_pool_listeners(self, service, pool_id):
-        pools_listeners = []
-        for listener in service['listeners']:
-            if listener['default_pool_id'] == pool_id:
-                pools_listeners.append(listener)
-        return pools_listeners
+                else:
+                    pool['provisioning_status'] = plugin_const.ACTIVE
+                    pool['operating_status'] = lb_const.ONLINE
```

Comment:
Same as for listener: pool can be created with admin_state_up = False, and then should have an operating_status of OFFLINE.